src/Angor/Client/Pages/View.razor

```diff
@@ -454,27 +516,97 @@
             notificationComponent.ShowNotificationMessage("You must create a wallet if you want to invest");
     }
 
-    private async Task CopyNsecToClipboardAsyncCheckPassword()
+
+    private async Task ShowNsecAsync()
     {
         if (!passwordComponent.HasPassword())
         {
-            passwordComponent.ShowPassword(CopyNsecToClipboardAsync);
+            passwordComponent.ShowPassword(async () =>
+            {
+                await GenerateNsec();
+            });
         }
         else
         {
-            await CopyNsecToClipboardAsync();
+            await GenerateNsec();
+        }
+    }
+
+    private async Task GenerateNsec()
+    {
+        isGeneratingNsec = true;
+        errorMessage = string.Empty;
+        StateHasChanged();
+
+        try
+        {
+            if (project is FounderProject founderProject)
+            {
+                var words = await passwordComponent.GetWalletAsync();
+                var nostrKey = _derivationOperations.DeriveProjectNostrPrivateKey(words, founderProject.ProjectIndex);
+                var nsec = NBitcoin.DataEncoders.Encoders.Hex.EncodeData(nostrKey.ToBytes());
+                NostrSecKey = nsec;
+                StateHasChanged();
+            }
+        }
+        catch (InvalidOperationException)
+        {
```

Comment:
should we not handle the password check on the popup when entering the password?
I think we do or used to do it.

Reply:
Yes, it is checked, but if the user does not press the submit button and exits the password entry window by pressing X, then we can no longer check the wrongly entered password.